File: 2024_fase1/solutions/linguica.c
#include <stdio.h>

int main(){
    int N;
    scanf("%d", &N);

    int P[N+1];
    for(int i=1; i<=N; i++)
      scanf("%d", P+i);

    int maxHeap = 1, minHeap = 1;

    for(int i=N; i>1; i--){
      if(P[i] > P[i/2])
        maxHeap = 0;
      if(P[i] < P[i/2])
        minHeap = 0;
    }

    if(maxHeap && minHeap)
      printf("0\n");
    else if(maxHeap)
      printf("2\n");
    else if(minHeap)
      printf("1\n");
    else
      printf("0\n");
    return 0;
}

File: 2024_fase1/solutions/linguica.py
N = int(input())

P = [0] + [int(input()) for _ in range(N)]

maxHeap = True
minHeap = True

for i in range(N, 1, -1):
    if P[i] > P[i//2]:
        maxHeap = False
    if P[i] < P[i//2]:
        minHeap = False

if maxHeap and minHeap:
    print(0)
elif maxHeap:
    print(2)
elif minHeap:
    print(1)
else:
    print(0)
